src/test/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewerSpec.groovy

```diff
@@ -0,0 +1,235 @@
+        }
+
+        and: 'Configuration for container name'
+        Holders.flatConfig.put('fileuploader.groups.SomeFileGroup.container', 'dummy-container')
+
+        when:
+        DefaultUFileTemporaryUrlRenewer renewer = new DefaultUFileTemporaryUrlRenewer(
+                CDNProvider.GOOGLE,
+                fileUploader,
+                forceAll,
+                10
+        )
+        renewer.renew()
+
+        then: 'No exception should be thrown'
+        noExceptionThrown()
+
+        where: 'The test data is as follows'
+        forceAll | cdnProviderThrowsException
+        false    | true
+        false    | false
+        true     | true
+        true     | false
+
+        description << [
+                'forceAll is false and getTemporaryURL throws an Exception',
+                'forceAll is false and getTemporaryURL does not throw an Exception',
+                'forceAll is true and getTemporaryURL throws an Exception',
+                'forceAll is true and getTemporaryURL does not throw an Exception'
+        ]
+    }
+
+    void "test that getResultListFromOffset method returns the correct instances for the given offset"() {
+        given: 'UFile instances'
+        List<UFile> uFiles = []
+        12.times { uFiles.add(createUFileInstance(new Date() - 100)) }
+
+        and: 'Mocked CDNFileUploader instance'
+        CDNFileUploader fileUploader = Mock(GoogleCDNFileUploaderImpl)
+
+        when: 'getResultListFromOffset method is called'
+        List<UFile> resultUFiles = new DefaultUFileTemporaryUrlRenewer(
+                CDNProvider.GOOGLE,
+                fileUploader,
+                true,
+                5
+        ).getResultListFromOffset(5)
+
+        then: 'The returned resultUFiles must matches with the expected list of UFiles'
+        resultUFiles == uFiles[5..9]
+    }
+
+    void "test that processResultList method updates the expiration date and url for the given list of UFiles"() {
+        given: 'UFile instances'
+        List<UFile> uFiles = []
+        5.times { uFiles.add(createUFileInstance(new Date() - 10)) }
+
+        and: 'Mocked CDNFileUploader instance'
+        CDNFileUploader fileUploader = Mock(GoogleCDNFileUploaderImpl)
+        fileUploader.getTemporaryURL(_, _, _) >> { String containerName, String fileName, long expiration ->
+            return 'Dummy Url'
+        }
+
+        and: 'Configuration for container name'
+        Holders.flatConfig.put('fileuploader.groups.SomeFileGroup.container', 'dummy-container')
+
+        when: 'processResultList method is called'
+        new DefaultUFileTemporaryUrlRenewer(
+                CDNProvider.GOOGLE,
+                fileUploader,
+                true,
+                5
+        ).processResultList(uFiles)
+
+        then: 'path and expiresOn properties for the given files should be updated'
+        uFiles.every { UFile uFile ->
+            return uFile.path == 'Dummy Url' && uFile.expiresOn > new Date()
```

Comment:
Can't we use assert here?

Reply:
Why do we need to use assert? This works.

Reply:
I am not sure if the condition is being validated. Maybe you can try to change it to `uFile.path != 'Dummy Url'` and see if the test fails. Just to be sure.

Reply:
The condition is being evaluated as we are using `every`. If we had used `each` then your doubt would have become real. 😃